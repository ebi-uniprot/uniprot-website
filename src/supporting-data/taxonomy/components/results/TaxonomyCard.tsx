import { Card } from 'franklin-sites';
import { FC, useCallback } from 'react';
import { useHistory } from 'react-router-dom';

import TaxonomyView from '../../../../shared/components/entry/TaxonomyView';
import RenderColumnsInCard from '../../../../shared/components/results/RenderColumnsInCard';

import { getEntryPath } from '../../../../app/config/urls';
import { getIdKeyFor } from '../../../../shared/utils/getIdKeyForNamespace';

import { Namespace } from '../../../../shared/types/namespaces';
import { TaxonomyAPIModel } from '../../adapters/taxonomyConverter';
import TaxonomyColumnConfiguration, {
  TaxonomyColumn,
} from '../../config/TaxonomyColumnConfiguration';

const lineage = TaxonomyColumnConfiguration.get(TaxonomyColumn.lineage);

const getIdKey = getIdKeyFor(Namespace.taxonomy);

const TaxonomyCard: FC<{
  data: TaxonomyAPIModel;
  selected?: boolean;
  handleEntrySelection: (rowId: string) => void;
}> = ({ data, selected, handleEntrySelection }) => {
  const history = useHistory();

  const id = getIdKey(data);

  const handleCardClick = useCallback(() => {
    history.push(getEntryPath(Namespace.taxonomy, id));
  }, [history, id]);

  return (
    <Card onClick={handleCardClick}>
      <div className="result-card">
        <div className="result-card__left">
          <input
            type="checkbox"
            checked={selected}
            onClick={(e) => e.stopPropagation()}
            onChange={() => handleEntrySelection(id)}
            data-testid="up-card-checkbox"
          />
        </div>
        <div className="result-card__right">
          <h5>
            <TaxonomyView data={data} />
          </h5>
          <RenderColumnsInCard data={data} renderers={lineage} />
        </div>
      </div>
    </Card>
  );
};

export default TaxonomyCard;
