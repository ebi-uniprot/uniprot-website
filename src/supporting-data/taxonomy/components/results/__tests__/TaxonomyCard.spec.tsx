import renderWithRouter from '../../../../../shared/__test-helpers__/customRender';
import TaxonomyCard from '../TaxonomyCard';

import mockData from '../../../__mocks__/taxonomyModelData';

describe('Taxonomy Card tests', () => {
  it('should render', () => {
    const { asFragment } = renderWithRouter(
      <TaxonomyCard data={mockData[0]} handleEntrySelection={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
