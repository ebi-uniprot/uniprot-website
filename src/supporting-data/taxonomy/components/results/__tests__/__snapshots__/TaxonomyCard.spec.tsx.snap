// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taxonomy Card tests should render 1`] = `
<DocumentFragment>
  <section
    class="card"
  >
    <div
      class="card--has-hover"
      role="button"
      tabindex="0"
    >
      <div
        class="card__content"
      >
        <section
          class="result-card"
        >
          <div
            class="result-card__left"
          >
            <input
              data-testid="up-card-checkbox"
              type="checkbox"
            />
          </div>
          <div
            class="result-card__right"
          >
            <h5>
              <a
                href="/taxonomy/11652"
              >
                Primate lentivirus group 
              </a>
            </h5>
            <div
              class="result-card__info-container"
            >
              <span
                class="result-card__info-bit"
              >
                <strong>
                  Lineage
                </strong>
                : 
                <a
                  href="/taxonomy/10239"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/2559587"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/2732397"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/2732409"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/2732514"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/2169561"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/11632"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/327045"
                >
                   
                </a>
                 &gt; 
                <a
                  href="/taxonomy/11646"
                >
                   
                </a>
              </span>
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
